master/internal/experiment.go

```diff
@@ -199,22 +193,14 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 	case trialCreated:
 		ops, err := e.searcher.TrialCreated(msg.create, msg.trialID)
 		e.processOperations(ctx, ops, err)
-	case trialCompletedOperation:
-		ops, err := e.searcher.OperationCompleted(msg.trialID, msg.op, msg.metrics)
+	case trialReportValidation:
+		ops, err := e.searcher.ValidationCompleted(msg.trialID, msg.metrics)
 		e.processOperations(ctx, ops, err)
-	case trialCompletedWorkload:
-		for _, op := range msg.completedOps {
-			ops, err := e.searcher.OperationCompleted(msg.trialID, op.op, op.metrics)
-			e.processOperations(ctx, ops, err)
-		}
-		if msg.earlyExit != nil {
-			ops, err := e.searcher.TrialExitedEarly(msg.trialID, msg.earlyExit.exitedReason)
-			e.processOperations(ctx, ops, err)
-		}
-	case trialValidation:
-		if msg.validationMetrics != nil {
-			ctx.Respond(e.isBestValidation(*msg.validationMetrics))
-		}
+	case trialReportEarlyExit:
+		ops, err := e.searcher.TrialExitedEarly(msg.trialID, msg.reason)
+		e.processOperations(ctx, ops, err)
+	case trialQueryIsBestValidation:
```

Comment:
can this be a field in `trialReportValidation`? we could handle the logic of saving the best there.

Reply:
probably, with out it's written it's sort of a chicken/egg situation, `sequencer.WorkloadCompleted` needs to know if it was the best validation and it tells us whether to report the validation to the searcher, so the call ordering has to be ask trialQueryIsBestValidation -> sequencer.WorkloadCompleted -> maybe tell trialReportValidation. I could untie this but since @rb-determined-ai is rewriting the sequencer in python, I figured I may as well wait rather than refactoring code that'll be removed shortly.